File: FlickrApp/Cells/ImageCell.m
//
//  ImageCell.m
//  FlickrApp
//
//  Created by aSqar on 23.11.2017.
//  Copyright © 2017 Askar Bakirov. All rights reserved.
//

#import "ImageCell.h"
#import "ImageViewModel.h"
#import <SDWebImage/UIImageView+WebCache.h>

@interface ImageCell ()

@property (nonatomic, weak) IBOutlet UIImageView *imgPhoto;
@property (nonatomic, weak) IBOutlet UILabel *lblName;
@property (nonatomic, weak) IBOutlet UIProgressView *progressView;

@end

@implementation ImageCell

- (void) setViewModel:(ImageViewModel *)viewModel
{
    _viewModel = viewModel;
    
    [_imgPhoto sd_setImageWithURL: viewModel.url placeholderImage:[UIImage imageNamed:@"placeholder.png"]
     options:0 progress:^(NSInteger receivedSize, NSInteger expectedSize){
         [self.progressView setProgress: receivedSize / expectedSize];
     } completed:^(UIImage *image, NSError *error, SDImageCacheType cacheType, NSURL *imageURL){
         self.progressView.hidden = YES;
    }];
    
    _lblName.text = viewModel.caption;
}

@end

File: FlickrApp/Cells/ImageCell.swift
//
//  ImageCell.swift
//  FlickrApp
//
//  Created by aSqar on 23.11.2017.
//  Copyright © 2017 Askar Bakirov. All rights reserved.
//

import UIKit
import SDWebImage

class ImageCell : UICollectionViewCell {
    
    @IBOutlet weak var imgPhoto: UIImageView?
    @IBOutlet weak var lblName: UILabel?
    @IBOutlet weak var progressView: UIProgressView?
    
    var viewModel:ImageViewModel!

    func setViewModel(viewModel:ImageViewModel!) {
        self.viewModel = viewModel

        imgPhoto?.sd_setImage(with: viewModel.url(isThumbnail:true), placeholderImage:UIImage(named: "placeholder.png"),
                                    options:SDWebImageOptions(rawValue: 0), progress:{ (receivedSize,expectedSize,url) in
                                        DispatchQueue.main.async {
                                            self.progressView!.progress = Float(receivedSize / expectedSize)
                                        }
         }, completed:{ (image,error,cacheType,imageURL) in
            self.progressView!.isHidden = true
            })

        lblName?.text = viewModel.caption
    }
}
